File: src/main/java/com/rephilo/luandun/leetcode/_122/Solution.java
package com.rephilo.luandun.leetcode._122;

public class Solution {
    /**
     * 122 只要后一天比前一天高，就可以按照一天一天的买进
     *
     * @param prices
     * @return
     */
    public int maxProfit(int[] prices) {
        int max = 0;
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] - prices[i - 1] > 0) {
                max += prices[i] - prices[i - 1];
            }
        }

        return max;
    }
}

File: src/main/scala/com/rephilo/luandun/leetcode/_122/Solution.scala
package com.rephilo.luandun.leetcode._122

object Solution {
  /**
    * 122
    * sliding函数（选取的元素数，步长）
    *
    * @param prices
    * @return
    */
  def maxProfit(prices: Array[Int]): Int = {
    if (prices.length < 2) 0 else prices.sliding(2, 1).map(x => x(1) - x(0)).filter(_ > 0).sum
  }
}
